### ISSUE TITLE
conditional on the main variable

### ISSUE THREAD

**Issue by DimitrisVoul** [nlmixrdevelopment/nlmixr#188]
Hello, 

I am using nlmixr and try to calculate a volume V. I want to have an IF statement that uses a condition on V to change the equation. For example:

```
if (V>=1.5) {
d/dt(V) = k*V-d
 } else {
d/dt(V) = k*V
}
```

When I do that I get the following error:

> Calculate ETA-based prediction and error derivatives:
> ################################################################################
> ## Condition: (!(V>=1.5))
> ################################################################################
> Calculate Jacobian
> [====|====|====|====|====|====|====|====|====|====] 100%;  0.724 sec 
> Calculate Sensitivities
> [====|====|====|====|====|====|====|====|====|====] 100%;  1.667 sec 
> Load into sympy...done
> Calculate d(f)/d(eta)
> Error in rxSymPy(sprintf("diff(%s,%s)", prd, state)) : 
>   Error running SymPy command:
>     diff(rx_pred_,S)
>     rx_pred_: V
> 
> Traceback (most recent call last):
>   File "<string>", line 3, in <module>
>   File "c:\R\NLMIXR~1.0-9\R\rxPython\Lib\SITE-P~1\sympy\core\function.py", line 2312, in diff
>     return f.diff(*symbols, **kwargs)
>   File "c:\R\NLMIXR~1.0-9\R\rxPython\Lib\SITE-P~1\sympy\core\expr.py", line 3095, in diff
>     return Derivative(self, *symbols, **assumptions)
>   File "c:\R\NLMIXR~1.0-9\R\rxPython\Lib\SITE-P~1\sympy\core\function.py", line 1283, in __new__
>     if not v._diff_wrt:
>   File "c:\R\NLMIXR~1.0-9\R\rxPython\Lib\SITE-P~1\sympy\core\singleton.py", line 113, in __getattr__
>     name, self))
> AttributeError: Attribute '_diff_wrt' was not installed on SymPy registry S
> Freeing Python/SymPy memory...done
> 
> 

Not sure what the issue is. Is that allowed in FOCEi?

**Comment by mattfidler** [comment]
It should be allowed in FOCEi,  Can you post the full model and a dummy dataset (even if it is simply applying it to `theo_sd`_

**Comment by DimitrisVoul** [comment]
I realised that there must be an error somewhere else/ I try to run a mode; with a main volume and two sub volumes (V1+V2=V) adn we assume that the data we have are for the main volume V:

`one.compartment <- function() {
  ini({
    tk1 <- 0.45 
    tk2 <- 1 
    tk3 <- 3.45  
    tV0 <- -1.43

    eta.k1 ~ 0.6
    eta.k2 ~ 0.3
    eta.k3 ~ 0.1
    eta.V0 ~ 0.07
    prop.err <- 0.7
  })
  model({
    k1 <- exp(tk1 + eta.k1)
    k2 <- exp(tk2 + eta.k2)
    k3 <- exp(tk3 + eta.k3)
    V0<- exp(tV0+eta.V0)
    
    V(0)<- V0
    V1(0)<-0.3*V0
    V2(0)<-0.7*V0
    
    d/dt(V) = k1*V1+k2*V2
    d/dt(V1) = k1 * V1 -0.5*V1
    d/dt(V2) = k2 * V2 +0.5*V2
    


    V ~ prop(prop.err)
  })
}

fitF <- nlmixr(one.compartment, theo_sd, est="focei")
`
When I try to run this R say ETA[4] is missing and then crashes. I must be unaware of how to set up a model like this.

**Comment by DimitrisVoul** [comment]
I found the error. it was because I had a variable with name S which clashed with something in nlmixr. Solving that allowed it to run with a conditional on V. My question now is, is it possible to have more than one conditional? i.e

> if (V>4 | (TIME>=10 & V>2) ) {
> ...
> }

?

**Comment by mattfidler** [comment]
In theory it should be fine.  

The `S` may have been corrected, but I can check later.

**Comment by mattfidler** [comment]
Did you figure this out or can you produce a reproducible example I can look at?

**Comment by DimitrisVoul** [comment]
Hello,

Sorry I was away on annual leave. So it seems to work but I also found another way around it.

Since the conditional are there to change the form of the differential equation depending on the conditions what I did was to create columns in the dataset that take the value of 0 or 1 depending on whether the condition is satisfied. In the model section then the parameters that depend on the conditions are multiplied by the value of the respective column and are hence zero or whatever value was set.
